app/src/main/java/com/github/fribourgsdp/radio/LyricsGetter.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.fribourgsdp.radio
+
+import android.util.Log
+import okhttp3.*
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+import org.json.JSONObject
+import java.util.concurrent.ExecutionException
+
+/**
+ * Tool to get lyrics from a given song name and artist using Musixmatch API.
+ * API Call doc : https://stackoverflow.com/questions/45219379/how-to-make-an-api-request-in-kotlin
+ */
+
+private const val API_KEY = "a3454edb65483e706c127deaa11df69d"
+private const val BASE_URL = "http://api.musixmatch.com/ws/1.1/"
+private const val LYRICS_NOT_FOUND = "---No lyrics were found for this song.---"
+
+
+
+class LyricsGetter {
+
+    companion object {
+
+
+        fun getLyrics(songName: String, artistName: String, client: OkHttpClient) : CompletableFuture<String> {
+            val future = CompletableFuture<String>()
+            val trackIDFuture = getSongID(songName, artistName, client)
+            val trackID : Int
+            try {
+                trackID = trackIDFuture.get()
+            } catch (e : Throwable){
+                future.complete(LYRICS_NOT_FOUND)
+                return future
+            }
+
+            val url = BASE_URL + "track.lyrics.get?track_id=" + trackID + "&apikey=" + API_KEY
+            val request = Request.Builder().url(url).build()
+            client.newCall(request).enqueue(object : Callback {
+                override fun onFailure(call: Call, e: IOException) {
+                    future.completeExceptionally(e)
+                }
+
+                override fun onResponse(call: Call, response: Response) {
+                    val parsedResponseString = response.body()?.string()
+                    val parsedResponse = parsedResponseString?.let { JSONObject(it) }
```

Comment:
I think this call should consider the fact JSONObject() may throw an exception if the JSON parsing isn't successful. If no exception is thrown, then you shouldn't need to check if it's null or not.

Reply:
good idea